File: lista Go/at7.go
package main

import "fmt"


type Professor struct {
    Nome    string
    Escolas []*Escola  
}


func (p *Professor) AdicionarEscola(escola *Escola) {
    p.Escolas = append(p.Escolas, escola)
}


func (p Professor) ExibirEscolas() {
    fmt.Printf("Professor %s leciona nas escolas:\n", p.Nome)
    for _, escola := range p.Escolas {
        fmt.Println(" -", escola.Nome)
    }
}


type Escola struct {
    Nome       string
    Professores []*Professor  
}


func (e *Escola) AdicionarProfessor(professor *Professor) {
    e.Professores = append(e.Professores, professor)
}


func (e Escola) ExibirProfessores() {
    fmt.Printf("Escola %s tem os professores:\n", e.Nome)
    for _, professor := range e.Professores {
        fmt.Println(" -", professor.Nome)
    }
}

func main() {
   
    professor1 := &Professor{Nome: "Carlos"}
    professor2 := &Professor{Nome: "Maria"}

    
    escola1 := &Escola{Nome: "Escola A"}
    escola2 := &Escola{Nome: "Escola B"}

    
    escola1.AdicionarProfessor(professor1)
    escola1.AdicionarProfessor(professor2)
    escola2.AdicionarProfessor(professor2)

    
    professor1.AdicionarEscola(escola1)
    professor2.AdicionarEscola(escola1)
    professor2.AdicionarEscola(escola2)

    
    escola1.ExibirProfessores()
    escola2.ExibirProfessores()

    
    professor1.ExibirEscolas()
    professor2.ExibirEscolas()
}

File: lista java/at7.java
import java.util.ArrayList;
import java.util.List;


class Professor {
    private String nome;
    private List<Escola> escolas = new ArrayList<>();  

   
    public Professor(String nome) {
        this.nome = nome;
    }

    
    public void adicionarEscola(Escola escola) {
        escolas.add(escola);
    }

    
    public void exibirEscolas() {
        System.out.println("Professor " + nome + " leciona nas escolas:");
        for (Escola escola : escolas) {
            System.out.println(" - " + escola.getNome());
        }
    }

    public String getNome() {
        return nome;
    }
}


class Escola {
    private String nome;
    private List<Professor> professores = new ArrayList<>();  

    
    public Escola(String nome) {
        this.nome = nome;
    }

    
    public void adicionarProfessor(Professor professor) {
        professores.add(professor);
    }

   
    public void exibirProfessores() {
        System.out.println("Escola " + nome + " tem os professores:");
        for (Professor professor : professores) {
            System.out.println(" - " + professor.getNome());
        }
    }

    public String getNome() {
        return nome;
    }
}

public class Main {
    public static void main(String[] args) {
        
        Professor professor1 = new Professor("Carlos");
        Professor professor2 = new Professor("Maria");

        
        Escola escola1 = new Escola("Escola A");
        Escola escola2 = new Escola("Escola B");

        escola1.adicionarProfessor(professor1);
        escola1.adicionarProfessor(professor2);
        escola2.adicionarProfessor(professor2);

        
        professor1.adicionarEscola(escola1);
        professor2.adicionarEscola(escola1);
        professor2.adicionarEscola(escola2);

        
        escola1.exibirProfessores();
        escola2.exibirProfessores();

        
        professor1.exibirEscolas();
        professor2.exibirEscolas();
    }
}
